content/blog/2022/2022-01-25-annotation-processing.md

```diff
@@ -0,0 +1,951 @@
+
+        Manager manager = new Manager();
+        manager.getemployeeStatus();
+    }
+}
+```
+
+We now add the annotation @Override to the 'getemployeeStatus' method and we get a compile-time error, which forces us to correct the typo right away.
+
+```java
+public class Manager extends Employee {
+
+    @Override
+    public void getemployeeStatus(){
+
+        System.out.println("This is the Manager class");
+    }
+}
+
+.\com\reflectoring\Manager.java:5: error: method does not override or implement a method from a supertype
+    @Override
+    ^
+1 error
+
+```
+
+### @FunctionalInterface 
+
+**Use Case for @FunctionalInterface** - It is used to indicate that the interface cannot have more than one abstract method. The compiler throws an error in case there is more than one abstract method. Functional Interfaces were introduced in Java 8, to implement Lambda expressions and ensure that they didn't make use of more than one method. Even without the @FunctionalInterface annotation, the compiler will throw an error if you include more than one abstract method in the interface. So why do we need @FunctionalInterface if it is not mandatory? 
+
+Let us take the example of the code below:
+
+```java
+
+@FunctionalInterface
+interface Print {
+    void printString(String testString);
+}
+
+public class FunctionalInterfaceTest {
+
+    public static void main(String args[]) {
+        Print testPrint = (String testString) -> System.out.println(testString);
+        testPrint.printString("This is a String");
+    }
+}
+```
+
+If you add another method, 'printString2', to the Print interface, the compiler or the IDE will throw an error and this will be obvious right away. Now, what if the Print interface was in a separate module, and there was no @FunctionalInterface annotation. Some other developers could easily add another function to the interface. Now if someone try to implement a Lambda function using the interface, it would break the code. Further, now we have to figure out which was the right function. By adding the @FunctionalInterface notation we get an immediate warning in the IDE, such as:
+
+​	**Multiple non-overriding abstract methods found in interface com.reflectoring.Print**
+
+So it is good practice to always include the @FunctionalInterface.
+
+### @SafeVarargs
+
+**Use Case for @SafeVarags** - The varargs functionality allowed the creation of methods with variable arguments. Prior to Java 5, the only option to creating methods with optional parameters was to create multiple methods, each with a different number of parameters. Varargs allowed us to create a single method to handle optional parameters with syntax as below:
+
+```
+void printStrings(String... stringList)
+
+instead of ....void printStrings(String string1, String string2)
+```
+
+However, warnings were thrown when generics were used in the arguments. @SafeVarargs allowed for suppression of these warnings. For eg., consider the code below:
+
+```
```

Comment:
Please make sure that code examples are always marked with the language, `java` in this case, otherwise they won't be syntax-highlighted in the article